cl/users/views.py

```diff
@@ -828,3 +833,37 @@ class RateLimitedPasswordResetView(PasswordResetView):
     template_name = "register/password_reset_form.html"
     email_template_name = "register/password_reset_email.html"
     form_class = CustomPasswordResetForm
+
+
+class SafeRedirectLoginView(auth_views.LoginView):
+    """
+    Custom LoginView that validates and sanitizes the redirect URL after a
+    successful login.
+
+    This view inherits from Django's built-in LoginView but adds an extra layer
+    of security by ensuring the redirect URL submitted by the login form is safe
+    It prevents potential open redirect vulnerabilities.
+    """
+
+    def get_redirect_url(self):
+        """
+        Return the user-originating redirect URL if it's safe. otherwise falls
+        back to the default.
+
+        This method ensures users cannot be redirected to malicious URLs after
+        logging in, even if they attempt to provide one.
+        """
+        redirect_to = self.request.POST.get(
+            self.redirect_field_name,
+            self.request.GET.get(self.redirect_field_name),
+        )
+        if not redirect_to:
+            return settings.LOGIN_REDIRECT_URL
+
+        try:
+            cleaned_url = parse_url_with_ada(redirect_to)
+        except ValueError:
+            return settings.LOGIN_REDIRECT_URL
+
+        is_safe = is_safe_url(redirect_to, self.request)
+        return cleaned_url if is_safe else settings.LOGIN_REDIRECT_URL
```

Comment:
With my changes, this becomes, I think:

```suggestion
        return get_redirect_or_abort(self.request, self.redirect_field_name)
```

I think you can even slim down your comments, since they should be explained adequately by the one function you're running.